docs/docker.md

```diff
@@ -0,0 +1,48 @@
+# Run a Function in a Docker Container
+
+To run your function in a container, create a `Dockerfile` with the following contents:
+
+```Dockerfile
+# Use the official Node.js 10 image.
+# https://hub.docker.com/_/node
+FROM node:10
+# Create and change to the app directory.
+WORKDIR /usr/src/app
+# Copy application dependency manifests to the container image.
+# A wildcard is used to ensure both package.json AND package-lock.json are copied.
+# Copying this separately prevents re-running npm install on every code change.
+COPY package.json package*.json ./
+# Install production dependencies.
+RUN npm install --only=production
+# Copy local code to the container image.
+COPY . .
+# Run the web service on container startup.
+CMD [ "npm", "start" ]
+```
+
+Start the container locally by running `docker build` and `docker run`:
+
+```sh
+docker build -t helloworld . && docker run -p 8080:8080 -it helloworld
+```
+
+Send requests to this function using `curl` from another terminal window:
+
+```sh
+curl localhost:8080
+# Output: Hello, World
+```
+
+## Deploy a Container
+
+You can deploy your containerized function to Cloud Run or any Knative-based environment by following the [Cloud Run quickstart](https://cloud.google.com/run/docs/quickstarts/build-and-deploy).
```

Comment:
Cloud Run or Cloud Run on GKE. Other knative instances are managed with `kubectl` or in the future `kn`. Instructions for knative in that case will also need a service.yaml.

Reply:
What would you like me to say instead of Cloud Run?
Does Cloud Run not include Cloud Run on GKE?

Can you suggest a change?

Reply:
This is for Cloud Run only: https://cloud.google.com/run/docs/quickstarts/build-and-deploy

This shows how to build and deploy an existing container image to Cloud Run on GKE: https://cloud.google.com/run/docs/quickstarts/prebuilt-deploy-gke

Deploying to Knative anywhere
https://knative.dev/docs/install/getting-started-knative-app/

Reply:
I've removed the mention of Cloud Run on GKE.